wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java

```diff
@@ -147,11 +151,14 @@ public Set<OAuthAuthenticatorDescriptor> getRegisteredAuthenticators(UriInfo uri
                   .withName("mode")
                   .withRequired(true)
                   .withDefaultValue("federated_login")));
-      OAuthAuthenticator authenticator = providers.getAuthenticator(name);
+      OAuthAuthenticator authenticator = oauth2Providers.getAuthenticator(name);
       result.add(
           newDto(OAuthAuthenticatorDescriptor.class)
               .withName(name)
-              .withEndpointUrl(authenticator.getEndpointUrl())
+              .withEndpointUrl(
+                  authenticator != null
+                      ? authenticator.getEndpointUrl()
+                      : oauth1Providers.getAuthenticator(name).getEndpointUrl())
```

Comment:
Optional: it might be cleaner to do something like
```java
OAuthAuthenticator authenticator = oauth2Providers.getAuthenticator(name);
if authenticator == null {
    authenticator = oauth1Providers.getAuthenticator(name)
}
//...
.withEndpointUrl(authenticator.getEndpointUrl())
```

Reply:
Right, of course -- I was getting confused with the name duplication :sweat_smile: